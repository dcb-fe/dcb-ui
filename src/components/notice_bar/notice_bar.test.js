import NoticeBar from './notice_bar.vue';
import { shallowMount } from '@vue/test-utils';

describe('NoticeBar', () => {
  describe('视图', () => {
    test('滚动内容', () => {
      const wrapper = shallowMount(NoticeBar, {
        propsData: {
          text: '你好多彩宝',
          leftIcon: 'notice',
          mode: 'link',
          wrapable: false,
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('逻辑', () => {
    // ...
  });
});
