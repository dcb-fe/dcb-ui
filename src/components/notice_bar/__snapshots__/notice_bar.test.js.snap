// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar 视图 时不渲染遮罩和内容 1`] = `
<div class="d-notice_bar" style="color: rgb(237, 106, 11); background-color: rgb(255, 251, 232);">
  <!---->
  <div class="d-notice_bar-notice_text">
    <div class="d-notice_bar-wrapable d-notice_bar-scrollX">
      你好多彩宝
    </div>
  </div>
  <!---->
</div>
`;

exports[`NoticeBar 视图 滚动内容 1`] = `
<div class="d-notice_bar" style="color: rgb(237, 106, 11); background-color: rgb(255, 251, 232);">
  <icon-stub name="notice" class="d-notice_bar-icon"></icon-stub>
  <div class="d-notice_bar-notice_text">
    <div class="d-notice_bar-wrapable d-notice_bar-scrollX">
      你好多彩宝
    </div>
  </div>
  <icon-stub name="return" class="d-notice_bar-right_icon"></icon-stub>
</div>
`;
